### PR TITLE
Embeddings: remove context detection index

### PR DESCRIPTION
**Pull request by jtibshirani**
Our context detection embeddings index adds complexity to the system:
* We need to make sure it's populated once when embeddings are configured. **Currently this is buggy ... for example it's totally missing if an admin sets an embeddings policy (but never manually schedules)!**
* When the embeddings model changes, admins will need to migrate it to the new model too. We'd need to add some automatic upgrade, or surface this index in the admin UI.

The complexity probably outweighs the benefit we're getting from it. LLMs do a good job of ignoring irrelevant context, so supplying context when it's not needed won't hurt response quality.

The main risk: chat fluency might get worse. We might be worse at detecting when a chat message is referencing the previous one, so it should use the existing context. I tried to mitigate this by expanding the "no context regexes" with a bunch of common cases I've encountered. 

## Test plan

Expanded context detection unit test. Manually tested short chats to check chat fluency is still okay:
* What is BrandLogo? (context) … That doesn’t seem right (no context)
* Is crewjam/saml used anywhere in the code? (context) … Explain in more detail (no context)
*  are sub-repo permissions respected in the embeddings service? (context) … I don’t understand (no context)

### CODE REVIEW

**Comment by jtibshirani** [comment]
I think we should dogfood this for a couple days to see if we're okay with it. For that reason, I haven't yet removed any DB tables or updated client code to not call `isContextRequiredForChatQuery`. Our acceptance criteria can be "no one notices or complains", or "1-2 people notice, but their issues are easily fixed".

File: cmd/frontend/graphqlbackend/embeddings.graphql [hunk 1]
```diff
@@ -92,10 +92,6 @@ extend type Mutation {
     """
     scheduleRepositoriesForEmbedding(repoNames: [String!]!, force: Boolean): EmptyResponse!
     """
-    Experimental: Schedules a job to create an embedding index used for context detection. The index is used to determine wheter a query requires additional context.
-    """
-    scheduleContextDetectionForEmbedding: EmptyResponse!
```
Comment (camdencheek) [inline cmd/frontend/graphqlbackend/embeddings.graphql:97]: What was calling this? Was this supposed to be manually called?

**Review by camdencheek — APPROVED** [review]
LGTM!

File: cmd/frontend/graphqlbackend/embeddings.graphql (hunk 1 above)
Comment (jtibshirani) [reply cmd/frontend/graphqlbackend/embeddings.graphql:97]: Yes! The client called this every time a new repo embedding job was scheduled:

https://github.com/sourcegraph/sourcegraph/blob/983b224c54744706e8f330d1a588d3407a6fac7b/client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx#L83-L86

So this is also a nice simplification / cost savings.

**Review by abeatrix — APPROVED** [review]
TY!

**Pull request merged**